gemrb/GUIScripts/GUICommonWindows.py

```diff
@@ -2042,12 +2044,14 @@ def UpdateClock ():
 
 	else:
 		Clock = None
-		if OptionsWindow and OptionsWindow.HasControl (9):
-			Clock = OptionsWindow.GetControl (9)
+		if OptionsWindow:
+			if GameCheck.IsIWD2():
+				Clock = OptionsWindow.GetControl (10)
+			elif OptionsWindow.HasControl(9):
+				Clock = OptionsWindow.GetControl (9)
 		elif ActionsWindow and ActionsWindow.HasControl (62):
 			Clock = ActionsWindow.GetControl (62)
-
-		if Clock and Clock.HasAnimation("CGEAR"):
+		if Clock and (Clock.HasAnimation("CGEAR") or GameCheck.IsIWD2()):
```

Comment:
Is it safe to assume Clock will always be non-null in iwd2? Eg., do all the window packs have it and at the same control id?

Reply:
All the `CGEAR`s I found in InfExp here were indexed by 9. I can double-check for positivity, and then I guess I'd had to find something calling `UpdateClock` without a clock but options. I just assumed this for now as I really couldn't find a counter-example by clicking through all reachable window packs.

Reply:
I claim that we are good to go here, for IWD2 ist options bar <=> clock. And for every found `GCOMM` bitmap, it's index 10.